src/features/posts.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+
+type PostsType = {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostsType = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    setPosts: (posts, { payload }: PayloadAction<Post[]>) => {
+      posts.items = [...payload];
+    },
+    setLoaded: (posts, { payload }: PayloadAction<boolean>) => {
+      posts.loaded = payload;
+    },
+    setError: (posts, { payload }: PayloadAction<boolean>) => {
+      posts.hasError = payload;
```

Comment:
It might be useful to include an error message in your state. This way, you can provide more specific feedback to the user when something goes wrong. You could change the `hasError` boolean to `errorMessage` string.